services/date/date.service.js

```diff
@@ -0,0 +1,48 @@
+'use strict'
+
+const BaseService = require('../base')
+const { formatRelativeDate } = require('../../lib/text-formatters')
+
+module.exports = class Date extends BaseService {
+  static render({ relativeDateString }) {
+    return {
+      message: relativeDateString,
+      color: relativeDateString === 'invalid date' ? 'red' : 'blue',
+    }
+  }
+
+  async handle({ timestamp }) {
+    return this.constructor.render({
+      relativeDateString: formatRelativeDate(timestamp),
+    })
+  }
+
+  // Metadata
+  static get defaultBadgeData() {
+    return { label: 'date' }
+  }
+
+  static get category() {
+    return 'other'
+  }
+
+  static get url() {
+    return {
+      base: 'date',
+      format: '([0-9]+)',
+      capture: ['timestamp'],
+    }
+  }
+
+  static get examples() {
+    return [
+      {
+        title: 'Relative date',
+        urlPattern: ':timestamp',
+        staticExample: this.render({ relativeDateString: '2 days ago' }),
+        exampleUrl: '1540814400000',
```

Comment:
I found it confusing that we're using a 13 digit (no of milliseconds) timestamp here instead of the more usual 10 digit unix format (no of seconds). We probably don't need millisecond precision for this application, so it might be worth going with the 10 digit format. What do you think?
Either way, we should probably pass a `'documentation'` key here which explains which format we use and maybe link to a useful converter.

Reply:
Sorry.. just realised that comment about a documentation key might not have been clear. There's an example of this here - we define a `docs` var here:

https://github.com/badges/shields/blob/52d642cf91d8587a3d664c960c10c173486ab9a5/services/steam/steam-workshop.service.js#L10-L22

and then pass it in a key called `'documentation'` when we define the examples:

https://github.com/badges/shields/blob/52d642cf91d8587a3d664c960c10c173486ab9a5/services/steam/steam-workshop.service.js#L181-L188

that renders it in the modal popup from the index page :)

Reply:
I feel like `ms` may make more sense as we are parsing it straight to `moment.js` rather than using `time * 1000` or `time + '000'`,
Although the end user wouldn't necessarily know we are using javascript.

Reply:
Yeah. We should definitely support the 10-digit format. If you wanted, we could do both. If we declare this route:

```js
static get route() {
  return {
    base: 'date',
    format: '([0-9]{13}|[0-9]{10})',
    capture: ['timestamp'],
  }
}
```

we'll capture either a 10 or 13-digit timestamp. Then we can process the timestamp appropriately based on length and we'll reject invalid formats. Everyone's happy.